src/bench/coin_selection.cpp

```diff
@@ -47,9 +47,12 @@ static void CoinSelection(benchmark::State& state)
         CAmount nValueRet;
         bool bnb_used;
         CoinEligibilityFilter filter_standard(1, 6, 0);
-        CoinSelectionParams coin_selection_params(false, 34, 148, CFeeRate(0), 0);
-        bool success = wallet.SelectCoinsMinConf(1003 * COIN, filter_standard, vCoins, setCoinsRet, nValueRet, coin_selection_params, bnb_used)
-                       || wallet.SelectCoinsMinConf(1003 * COIN, filter_standard, vCoins, setCoinsRet, nValueRet, coin_selection_params, bnb_used);
+        CoinSelectionParams coin_selection_params_pass_1(true, 34, 148, CFeeRate(0), 0);
+        bool success = wallet.SelectCoinsMinConf(1003 * COIN, filter_standard, vCoins, setCoinsRet, nValueRet, coin_selection_params_pass_1, bnb_used);
+        if (!success) {
+            CoinSelectionParams coin_selection_params_pass_2(false, 34, 148, CFeeRate(0), 0);
```

Comment:
sucess is never false, so this is dead code?

Reply:
You mean it should never be false as there is enough coins?

Reply:
Yes, otherwise the benchmark would currently fail, no?

Reply:
`success` can be false when BnB fails to find an exact match. The first pass should be using BnB (which is why it needs its own parameters with true set) which can fail. Then it falls back to not using BnB (which is the second params with false) which should never fail.

Reply:
But given the current input it always succeeds. I agree that a different test exercising that case would be nice.

Reply:
Ah, indeed. This should have different test cases to trigger both algorithms.